Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -197,9 +193,18 @@ private void CleanCache()
         /// <param name="entryName">The name of the entry</param>
         /// <param name="fileName">The name of the zip file on disk</param>
         /// <returns>A <see cref="Stream"/> of the appropriate zip entry</returns>
-        private Stream CreateStream(ZipFile zipFile, string entryName, string fileName)
+        private Stream CreateStream(CachedZipFile zipFile, string entryName, string fileName)
         {
-            var entry = zipFile.Entries.FirstOrDefault(x => entryName == null || string.Compare(x.FileName, entryName, StringComparison.OrdinalIgnoreCase) == 0);
+            ZipEntry entry;
+            if (entryName == null)
+            {
+                entry = zipFile.EntryCache.FirstOrDefault().Value;
```

Comment:
no need to change as it maintains existing behavior, but this should *probably* be implemented as `SingleOrDefault`.